app/src/main/java/com/duckduckgo/app/referral/AppInstallationReferrerStateListener.kt

```diff
@@ -155,11 +161,17 @@ class PlayStoreAppReferrerStateListener @Inject constructor(
     private fun referralResultReceived(result: ParsedReferrerResult) {
         referralResult = result
 
-        if (result is ReferrerFound) {
-            variantManager.updateAppReferrerVariant(result.campaignSuffix)
-            appReferrerDataStore.campaignSuffix = result.campaignSuffix
-
+        when (result) {
+            is CampaignReferrerFound -> {
+                variantManager.updateAppReferrerVariant(result.campaignSuffix)
+                appReferrerDataStore.campaignSuffix = result.campaignSuffix
+            }
+            is EuAuctionReferrerFound -> {
+                variantManager.updateAppReferrerVariant(VariantManager.RESERVED_EU_AUCTION_VARIANT)
+                appReferrerDataStore.installedFromEuAuction = true
+            }
```

Comment:
is it worth adding an `else` to deal with unexpected referrers?

Reply:
I did wonder about that, too, but decided not to since the `else` can mean a few different things (no referrer found, there was a problem parsing the referrer etc...) and I can't think of any single thing we'd want to do for each of them